Sources/Sentry/SentryTransport.m

```diff
@@ -0,0 +1,226 @@
+//
+//  SentryTransport.m
+//  Sentry
+//
+//  Created by Klemens Mantzos on 27.11.19.
+//  Copyright © 2019 Sentry. All rights reserved.
+//
+
+#if __has_include(<Sentry/Sentry.h>)
+
+#import <Sentry/SentryTransport.h>
+//#import <Sentry/SentryClient.h>
+#import <Sentry/SentrySDK.h>
+#import <Sentry/SentryClient+Internal.h>
+#import <Sentry/SentryLog.h>
+#import <Sentry/SentryDsn.h>
+#import <Sentry/SentryError.h>
+#import <Sentry/SentryUser.h>
+#import <Sentry/SentryQueueableRequestManager.h>
+#import <Sentry/SentryEvent.h>
+#import <Sentry/SentryNSURLRequest.h>
+#import <Sentry/SentryInstallation.h>
+#import <Sentry/SentryFileManager.h>
+#import <Sentry/SentryBreadcrumbTracker.h>
+#import <Sentry/SentryCrash.h>
+#import <Sentry/SentryOptions.h>
+#import <Sentry/SentryScope.h>
+#else
+#import "SentryTransport.h"
+//#import "SentryClient.h"
+#import "SentrySDK.h"
+#import "SentryClient+Internal.h"
+#import "SentryLog.h"
+#import "SentryDsn.h"
+#import "SentryError.h"
+#import "SentryUser.h"
+#import "SentryQueueableRequestManager.h"
+#import "SentryEvent.h"
+#import "SentryNSURLRequest.h"
+#import "SentryInstallation.h"
+#import "SentryFileManager.h"
+#import "SentryBreadcrumbTracker.h"
+#import "SentryCrash.h"
+#import "SentryOptions.h"
+#import "SentryScope.h"
+#endif
+
+@interface SentryTransport ()
+
+@property(nonatomic, strong) SentryFileManager *fileManager;
+@property(nonatomic, strong) id <SentryRequestManager> requestManager;
+@property(nonatomic, weak) SentryOptions *options;
+
+@end
+
+@implementation SentryTransport
+
+@synthesize sampleRate = _sampleRate;
+@synthesize maxEvents = _maxEvents;
+@synthesize maxBreadcrumbs = _maxBreadcrumbs;
+
+- (id)initWithOptions:(SentryOptions *)options {
+  if (self = [super init]) {
+      self.options = options;
+      NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+      NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
+      self.requestManager = [[SentryQueueableRequestManager alloc] initWithSession:session];
+
+      NSError* error = nil;
+      self.fileManager = [[SentryFileManager alloc] initWithDsn:options.dsn didFailWithError:&error];
+      if (nil != error) {
+          [SentryLog logWithMessage:(error).localizedDescription andLevel:kSentryLogLevelError];
+          return nil;
+      }
+      [self setupQueueing];
+  }
+  return self;
+}
+
+- (void)setSampleRate:(float)sampleRate {
+    if (sampleRate < 0 || sampleRate > 1) {
+        [SentryLog logWithMessage:@"sampleRate must be between 0.0 and 1.0" andLevel:kSentryLogLevelError];
+        return;
+    }
+    _sampleRate = sampleRate;
+    self.shouldSendEvent = ^BOOL(SentryEvent *_Nonnull event) {
+        return (sampleRate >= ((double)arc4random() / 0x100000000));
+    };
+}
```

Comment:
Please move this logic into the client, the transport should really just store/send events.

Reply:
done